monai/networks/layers/utils.py

```diff
@@ -47,6 +47,8 @@ def get_norm_layer(name: Union[Tuple, str], spatial_dims: Optional[int] = 1, cha
         kw_args["num_features"] = channels
     if has_option(norm_type, "num_channels") and "num_channels" not in kw_args:
         kw_args["num_channels"] = channels
+    if has_option(norm_type, "affine") and "affine" not in kw_args:
+        kw_args["affine"] = True
```

Comment:
this change will break some existing models... I think we should warn in this version with something like:

```suggestion
        warnings.warn(f"`affine` is not set in {norm_type}, in future releases the default will be changed to `affine=True`.")
```

and change it in v1.2

Reply:
This will break models. I had used instancenorm as the default because it gave better results and explicitly because it didn't have learnable parameters, it was a simple transform which reduced the number of parameters a network would need. We have an issue that, yes, a number of networks used instancenorm and its default of not having parameters so this will be a breaking change for tested models and those in the model zoo. I feel we shouldn't make this change now but we could introduce a "instancenorm_affine" item into the `Norm` factory to simplify doing this for future network definitions.